### PR TITLE
pp->x->hh common request (run3)

### PR DESCRIPTION
**Pull request by nanlu06**
make HH decay fragment without dipole recoil:

https://gitlab.cern.ch/cms-b2g/b-2-g-m-csample-requests/-/merge_requests/20

### CODE REVIEW

File: genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        pythia8PSweightsSettingsBlock,
+        processParameters = cms.vstring(
+            '24:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '24:onMode = off', # Turn off all W decays 
+            '24:onIfAny = 1 2 3 4 5', # Add W->qq decays 
+            '25:onMode = off', # Turn off all H decays 
+            '25:onIfMatch = 22 22', # Add H->gg decay
+            '25:onIfMatch = 24 -24', # Add H->WW decay
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py:23]: you don't want H->ZZ->4Q here

Comment (acarvalh) [reply genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py:23]: That follows the run2 request. To @wangjiannju @hqucms answer please

Comment (nanlu06) [reply genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py:23]: Thank you very much for the comment! I added genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G2Zto4Q_TuneCP5_13p6TeV_madgraph-pythia8.py for H->ZZ->4Q @menglu21  @acarvalh

**Comment by acarvalh** [comment]
Hi @menglu21 
Is there anything else preventing merge?
Thank you

File: genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G2L2Nu_TuneCP5_13p6TeV_madgraph-pythia8.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        pythia8CP5SettingsBlock,
+        pythia8PSweightsSettingsBlock,
+        processParameters = cms.vstring(
+            '15:onMode = on', # allow all tau decays. Leptonic and Hadronic 
+            '24:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '24:onMode = off', # Turn off all W decays 
+            '24:onIfAny = 11 13 15', # Add W->enu, W->munu, W->taunu 
+            '25:onMode = off', # Turn off all H decays 
+            '25:onIfMatch = 22 22', # Add H->gg decay
+            '25:onIfMatch = 24 -24', # Add H->WW decay
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G2L2Nu_TuneCP5_13p6TeV_madgraph-pythia8.py:24]: do you need H->ZZ just like 2B2V case

File: genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py (hunk 1 above)
Comment (menglu21) [reply genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py:23]: then maybe change the name to HHto2G2Wto4Q

**Comment by nanlu06** [comment]
Hi @menglu21 Thank you a lot for the suggestion. The name HHto2G4Q is changed to HHto2G2Wto4Q now. Is it OK to merge this PR?

File: genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G2L2Nu_TuneCP5_13p6TeV_madgraph-pythia8.py (hunk 2 above)
Comment (bbilin) [reply genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2G2L2Nu_TuneCP5_13p6TeV_madgraph-pythia8.py:24]: A kind reminder on this point. You might want to include Z -> l+l- Z->nunu similar to 2B2V2L2Nu case. 

Also please include 2V in the name for 2G samples, like 2G2V2L2Nu etc.

**Review by bbilin — CHANGES_REQUESTED** [review]

**Comment by nanlu06** [comment]
Hi @bbilin I submitted two decay fragments HHto2G2Vto2G4Q_TuneCP5_13p6TeV_madgraph-pythia8.py and HHto2G2Vto2G2L2Nu_TuneCP5_13p6TeV_madgraph-pythia8.py if I understand the request correctly.

**Comment by nanlu06** [comment]
Hi @menglu21 @bbilin can you take a look if this PR could be merged now? Many thanks.

**Pull request merged**